packages/jsapi-components/src/useSetPaddedViewportCallback.ts

```diff
@@ -26,9 +55,20 @@ export function useSetPaddedViewportCallback(
         getSize(table)
       );
 
-      table?.setViewport(first, last);
+      if (subscriptionRef.current == null) {
+        table?.setViewport(first, last);
+        return;
+      }
+
+      subscriptionRef.current?.update({
+        rows: {
+          first,
+          last,
+        },
+        columns: table?.columns ?? [],
```

Comment:
If table is not defined, I'm assuming there's no point in calling `update` on the subscription? If so, I would probably put a:
```typescript
if (table == null) {
  return
}
```
at the beginning of `setPaddedViewport` and don't bother with any of the logic if it's null. Can then also get rid of some `?` coalescing.